pkg/minishift/hostfolder/hostfolder.go

```diff
@@ -322,11 +323,11 @@ func mountCifsHostfolder(driver drivers.Driver, hostfolder *config.HostFolder) e
 	}
 
 	cmd := fmt.Sprintf(
-		"sudo mount -t cifs %s %s -o username=%s,password=%s",
+		"sudo mount -t cifs %s %s -o username=%s,password='%s'",
```

Comment:
We could split this already rather long method up a bit and extract for example a method which returns the command string, given some sensible input. This allows us to test everything up to the point where the command is actually executed. This way we can codify our expectations on how this command looks like when there are for example special characters in the password.

Reply:
I would rather see that happen as part of a coordinated refactor

Reply:
I also believe such a chance will unnecessarily conflict with the refactoring that happens in #317, such as commit: https://github.com/hferentschik/minishift/commit/718caf9403e8ece32adf7b42bbbc6f66c41aa333

Reply:
IMO we should have a separate issue for this. If @praveenkumar is up for it , he can fix both the issues at one PR.